src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -522,4 +521,47 @@ object DeltaTable {
     }
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Check if the provided `path` is the root of a Delta table using the given SparkSession.
+   *
+   * @since 0.4.0
+   */
+  @Evolving
+  def isDeltaTable(sparkSession: SparkSession, path: String): Boolean = {
+    DeltaTableUtils.isDeltaTable(sparkSession, new Path(path))
+  }
+
+  /**
+   * :: Evolving ::
+   *
+   * Check if the provided `path` string is the root of a Delta table.
+   *
+   * Note: This uses the active SparkSession in the current thread to search for the table. Hence,
+   * this throws error if active SparkSession has not been set, that is,
+   * `SparkSession.getActiveSession()` is empty.
+   *
+   * @since 0.4.0
+   */
+  @Evolving
+  def isDeltaTable(path: String): Boolean = {
+    val sparkSession = SparkSession.getActiveSession.getOrElse {
+      throw new IllegalArgumentException("Could not find active SparkSession")
+    }
+    isDeltaTable(sparkSession, path)
+  }
+
+  /**
+   * :: Evolving ::
+   *
+   * Check whether the provided table name is a Delta table based on information from the
+   * Catalog using the given SparkSession.
+   *
+   * @since 0.4.0
+   */
+  @Evolving
+  def isDeltaTable(spark: SparkSession, tableName: TableIdentifier): Boolean = {
```

Comment:
We don't officially support table definitions yet, so I'd remove this for now. Also TableIdentifier is an internal Spark API, therefore we shouldn't use it.